browser_history/generic.py

```diff
@@ -86,16 +94,23 @@ def __init__(self, plat: utils.Platform = None):
         if self.profile_support and not self.profile_dir_prefixes:
             self.profile_dir_prefixes.append("*")
 
-    def profiles(self) -> typing.List[str]:
-        """Returns a list of profile directories
+    def profiles(self,profile_file) -> typing.List[str]:
+        """Returns a list of profile directories. If the browser is supported on the current
+        platform but is not installed an empty list will be returned
+
         :rtype: list(str)
+
         """
+
+        if not os.path.exists(self.history_dir):
+            utils.logger.info('%s browser is not installed', self.name)
+            return []
         if not self.profile_support:
             return ["."]
         profile_dirs = []
         for files in os.walk(str(self.history_dir)):
             for item in files[2]:
-                if os.path.split(os.path.join(files[0],item))[-1] == self.history_file:
+                if os.path.split(os.path.join(files[0],item))[-1] == profile_file:
```

Comment:
What's the `profile_file` for?

Reply:
Since I'm using the same function to extract both Bookmarks and histories, I had to change the name, cause I couldnt use self.history_file anymore

Reply:
Ah okay